app/scripts/models/reliers/oauth.js

```diff
@@ -28,6 +28,8 @@ define(function (require, exports, module) {
   var SIGNIN_SIGNUP_QUERY_PARAM_SCHEMA = {
     access_type: Vat.accessType().renameTo('accessType'),
     client_id: Vat.clientId().required().renameTo('clientId'),
+    code_challenge: Vat.string().min(43).max(128),
+    code_challenge_method: Vat.string().allow('S256'),
```

Comment:
Do you *only* want to allow `S256`, or allow `S256` in addition to some stricter validation?

[from allow's docs](https://github.com/shane-tomlinson/vat/blob/master/docs/api.md#anyallowvalues):
Explicitly allow a value, before any conversion. Accepts one or more arguments. 

If you want an "only allow S256", use [`.valid`](https://github.com/shane-tomlinson/vat/blob/master/docs/api.md#anyvalidvalues).